src/org/mozilla/javascript/NativeObject.java

```diff
@@ -368,6 +375,61 @@ public Object execIdCall(IdFunctionObject f, Context cx, Scriptable scope,
                 }
                 return cx.newArray(scope, ids);
               }
+
+          case ConstructorId_entries:
+              {
+                  Object arg = args.length < 1 ? Undefined.instance : args[0];
+                  Scriptable obj = getCompatibleObject(cx, scope, arg);
+                  Object[] ids = obj.getIds();
+                  for (int i = 0; i < ids.length; i++) {
+                      if (ids[i] instanceof Integer) {
+                          Integer key = (Integer) ids[i]; 
+                          ids[i] = cx.newArray(scope, new Object[]{ ids[i], obj.get(key, scope) });
```

Comment:
I think the first element of the new Array should be a String, even when you need the integer index to do the lookup. Have you thought about implementing the [EnumerableOwnPropertyNames](https://tc39.es/ecma262/#sec-enumerableownpropertynames) abstract method and using it for keys, entries, and values?